### PR TITLE
Add shotgun menu after workspace reset/switch

### PR DESCRIPTION
**Pull request by riobo**
When resetting the current workspace or switching workspace, the shotgun menu disappear from the main menu bar.  The solution is to listen to the CuiMenusPostLoad notification sent from 3dsMax and add the menu.

### CODE REVIEW

**Comment by manneohrstrom** [comment]
@riobo I am totally slammed with work right now and @thebeeland is much more of a 3dsmax expert than I am, so I'll ask him to take a look!

**Comment by riobo** [comment]
@manneohrstrom @thebeeland, the notification "CuiMenusPostLoad" was not exposed to MaxPlus, so I had to expose it in 2018Update1.  So this new behavior will only work with Max2018 Update1 and latter.  There is an appropriate try/except that handles older Max versions when trying to register to the notification.

**Comment by riobo** [comment]
@thebeeland Please review.

File: engine.py [hunk 1]
```diff
@@ -143,15 +142,36 @@ def eventFilter(self, obj, event):
...
-        self._menu_generator.create_menu()
+        self._add_shotgun_menu()
 
-        self.tk_3dsmax.MaxScript.enable_menu()
+        try:
+            # Listen to the CuiMenusPostLoad notification in order to add
+            # our shotgun menu after workspace reset/switch.
+            MaxPlus.NotificationManager.Register(MaxPlus.NotificationCodes.CuiMenusPostLoad, self._on_menus_loaded)
+        except AttributeError:
+            self.log_debug("CuiMenusPostLoad notification code is not available in this version of MaxPlus.")
```
Comment (manneohrstrom) [inline engine.py]: nitpick: perhaps we could also offer a further explanation, something like: The Shotgun menu will not be updated when 3dsmax workspaces are switched.

**Pull request merged**